### DISCUSSION TITLE
Mod menu with maps/mods full name

### DISCUSSION THREAD

**Discussion by cyberbarabba** [Novum/vkQuake#657] [Ideas]
Hi, some engines show maps/mods full name instead of their folder name in the mod menu and it's very useful.
Is it possible to have it too in vkQuake? An editable list should be great to sort/name maps/mods in a preferred way (maybe based on a txt, cfg or whatever editable file).

**Comment by Novum** [comment]
No idea where that data would come from, there is as far as I know no standard for that.

**Comment by vide0hanz** [comment]
@cyberbarabba Which engines do this?

**Reply by cyberbarabba** [reply]
Irownwail and AfterQuake

**Comment by Novum** [comment]
I think Ironwail does, but I haven't looked yet. It might be a hardcoded list?

**Comment by Novum** [comment]
Yeah, it's just hardcoded:
```
static const char *const knownmods[][2] =
{
	{"id1",			"Quake"},
	{"hipnotic",	"Scourge of Armagon"},
	{"rogue",		"Dissolution of Eternity"},
	{"dopa",		"Dimension of the Past"},
	{"mg1",			"Dimension of the Machine"},
	{"q64",			"Quake (Nintendo 64)"},
	{"ctf",			"Capture The Flag"},
	{"udob",		"Underdark Overbright"},
	{"ad",			"Arcane Dimensions"},
};
```

**Reply by vide0hanz** [reply]
Ah, I figured as much. I personally just rename the directories. 

Since were on the topic, out of curiosity have you considered hard coding menus like this for the official episodes like the rerelease does? I'm all for a consistent and organized experience when it comes to official content but like, where do you draw the line between support/compatibility with the rerelease vs recreating it? The mods menu seems like a good middle ground imo.

**Comment by Novum** [comment]
The mod menu could definitely take some love. Ironwail even supports downloading from the Kex Quake mod server! Pretty cool.

On a side note, I sometimes wish @andrei-drexler wouldn't have forked QS another time, don't think it was necessary. Indirect drawing could have been implemented in Vulkan just as well and OpenGL is a dead end anyway in my opinion. But probably good to have some friendly competition :)

**Reply by vide0hanz** [reply]
I like the goal of squeezing every bit of performance out of slightly older hardware, but unfortunately for me the laptop I tried Ironwail on only supports OpenGL up to 3.1 so that was a dead end. Not that playing Quake on a 2011 Thinkpad with a trackpoint would fun time anyway lol.

vkQuake is my preferred sourceport though, and thus far every QOL feature implemented seems to be well thought out. I like the simplicity of it. Its got just enough modern touches while still allowing purists to play the game how they remembered it back in the day, in my opinion.

**Reply by Novum** [reply]
There are very few cards that can run OpenGL 4.0 but not Vulkan.

**Reply by cyberbarabba** [reply]
Yeah, mod menu is very useful and great. It needs to be enhanced by allowing custom naming/sorting for maps and prompting a skill selection for ones which don't have a start map, so it should be perfect! :)

**Reply by Malmanian** [reply]
Greetings! In IronWail you can also create a custom textfile named "**descript.ion**" and add it to the individual map/mod folder.

descript.ion will contain the map/mod name in the first row of the textfile that IronWail will read from, for the mods/maps not avail. in the hardcoded list

For example:

Quake\eoe\descript.ion:
	Epochs of Enmity
